Orm/Xtensive.Orm/Properties/Visibility.cs

```diff
@@ -29,4 +29,10 @@
 "7bb6356267f95e0b58d478cf56393c4f060fbaff42a9686272e37009ab71bfa2e41046e952f389" +
 "f37c6a033d1a2a5354fc97226fc469128e49e6a479ac5d1dd69d7da5607d0dc4ede0765d477745" +
 "1034dc3a15f1532d010db3e633e62fc5e67a3ed175457acb9dc6c9d39ccc8ecfdaae62df34d488" +
-"c45009b2")]
\ No newline at end of file
+"c45009b2")]
+[assembly: InternalsVisibleTo("Xtensive.Orm.Tracking, PublicKey=" +
+"00240000048000009400000006020000002400005253413100040000010001007f8790a0573a2a" +
+"4f7b77907240977b669d33774bcaef77ef39d5186de4d4c836623fb661be674c7db731bc376184" +
+"f6848fc6b64eeba506654c8212de624e9d1f15ad1fe765f769e9a5b59ecaab2866632a983e7828" +
+"f3bf20ff7afffa021951cf8adc9ac1e786716430b0637893fb71b7566405fe14acc5e340c4af4a" +
+"a293c994")]
```

Comment:
Extensions should not rely on internal members and should not be in friendly assemblies

Reply:
Even providers have no access to internals

Reply:
The alternative is - make `DirectSessionAccessor.GetChangedEntitiesInternal()` public
IMO it is better to move some extensions into main `Xtensive.Orm` project.

Reply:
> IMO it is better to move some extensions into main Xtensive.Orm project.

Why? What is the criteria for this movement?

Reply:
> Why? What is the criteria for this movement?

In this case `GetChangedEntitiesInternal()` optimization.

But in general integration of stable extensions into main project simplifies using DO.
Then we need to reference only 2 packages from the app:  Orm + DB driver (SqlServer or other)

Reply:
The idea of extension is that it is optional one-purpose package, if you need it then you install it, otherwise you don't. You look at this from your perspective when you 100% use it. That's biased openion.

Reply:
> The idea of extension is that it is optional one-purpose package, if you need it then you install it, otherwise you don't.

The same idea as for any function or class.
This `Xtensive.Orm.Tracking.dll` extension is just 16KB of binary. Not a big deal if the main package will be 16KB bigger and we avoid all installation efforts.

Reply:
I have changed implementation to use conditional compilation:
```
    public IEnumerable<EntityState> GetChangedEntities(PersistenceState persistenceState) =>
#if DO_SAFE_COLLECTION_WRAPPER
      Session.EntityChangeRegistry.GetItems(persistenceState);
#else
      Session.EntityChangeRegistry.GetContainer(persistenceState);
#endif
```
This allowd to have safe implementation for public DO package and optimized for our build (depending on environment variable)
And we need not  visibility-attributes and internal method anymore